packages/material-ui/src/Portal/Portal.js

```diff
@@ -65,6 +73,8 @@ Portal.propTypes = {
   disablePortal: PropTypes.bool,
   /**
    * Callback fired once the children has been mounted into the `container`.
+   *
+   * This prop will be deprecated and removed in v5, the ref can be used instead.
```

Comment:
```suggestion
   * You probably don't need this prop. Prefer to use callback refs on the component instead.
```

We haven't deprecated it yet.

Reply:
The motivation for the "will be deprecate" is to create a sentiment of urgency for somebody relying on the prop but has a use case (we are not aware of) that can't rely on the ref. Do you think that we should still go for the "probably not"? I'm taking the time to ask because it's a broder topic, not specific to this occurrence. I can remember that the same method is used for withWidth. We had people opening an issue "Please don't deprecated X" with some details consequent to the initial ultimatum.

Reply:
I could only find usage of it in Popper and that component has quite the complicated implementation for pretty simple behavior. I tried reducing the amount of logic to get a better hold of why we need onRendered instead of ref. 

One part of the issue is the misnomer. Makes it hard to mentally "click" what this prop does. 

I'd like to explore first why we need it in Popper and then we can go back to this one. As it looks like right now "you probably don't need it" is the better advise.


> The motivation for the "will be deprecate" is to create a sentiment of urgency for somebody relying on the prop

That is not the why but the what. Why do you want to create a sense of urgency?

Reply:
I had forgotten about the Popper. I believe that the child div ref could be used instead. Would it work?

> Why do you want to create a sense of urgency?

It incentives users to give us feedback **sooner** than if we wait to introduce the deprecation. it optimizes for learning. The sooner we have the information:
- the less likely we will introduce a deprecation that we have to revert
- the less likely somebody will use the prop and the better migration path going from v4 to v5

Reply:
Not sure what we gain by saying we will deprecate it if we're not sure about it. That can only hurt trust.

>  it optimizes for learning.

Could you elaborate on that?

Reply:
Sorry, I wasn't clear.

I think that we should first agree that removing the prop is the best long-term path.

Then, it's about waiting x months (x because we don't want people to feel too much code change pressure) before introducing the deprecation warning vs saying it right away and adding a warning only later on (it creates x months of additional learning opportunity).

I don't think that saying the prop will be deprecated and not deprecating it is worse than deprecating a prop and reverting. 

So, if you think that we are not ready yet to commit to a deprecation, your suggestion is better. If you think that we are good for deprecation, we have an opportunity to collect feedback x more months (optimize for "learning").